### PR TITLE
add build arg and build context args

### PR DESCRIPTION
**Pull request by khizunov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,6 +192,14 @@ env:
 env_file: path/to/env_file.env
 ```
 
+```yaml
+# Use the git revision as the build container tag
+# Allows to use the same build container unless the git revision changes
+# This is useful when using a CI system that caches the build container
+# Remember to commit if you changing the build container
```
Comment (ofir-amir) [inline README.md:199]:
```suggestion
# Remember to commit if you are changing the build container
```

**Review by ofir-amir — COMMENTED** [review]
The bundling of lint fixes and other fixes make it hard to read

**Comment by khizunov** [comment]
> The bundling of lint fixes and other fixes make it hard to read

Guilty, forgot to disable auto-format.

File: skipper.yaml [hunk 2]
```diff
@@ -1 +1 @@
-build-container-image: skipper-build
+build-container-image: skipper-build
```
Comment (ofir-amir) [inline skipper.yaml]: any reason for removal of the new line?

File: skipper/cli.py [hunk 3]
```diff
@@ -94,50 +108,29 @@ def build(ctx, images_to_build, container_context, cache):
...
+    build_args = (ctx.obj.get('build_args', ()) + (f'TAG={tag}',))
+    build_contexts = ctx.obj.get('build_contexts', ())
+    main_context = container_context or ctx.obj.get('container_context')
+
+    for image, dockerfile in valid_images_to_build.items():
+        utils.logger.info("Building image: %s", image)
+
+        options = BuildOptions(
+            Image(name=image, tag=tag, dockerfile=dockerfile),
+            main_context or os.path.dirname(dockerfile),
```
Comment (ofir-amir) [inline skipper/cli.py]: I would move line 116 to 120 and join the condition from 122.
Don't think that the optimization is that important

File: skipper/cli.py [hunk 4]
```diff
@@ -254,26 +247,29 @@ def run(ctx, interactive, name, env, publish, cache, command):
...
-                      net=ctx.obj['build_container_net'],
-                      publish=publish,
-                      volumes=ctx.obj.get('volumes'),
-                      workdir=ctx.obj.get('workdir'),
-                      use_cache=cache,
-                      workspace=ctx.obj.get('workspace'),
-                      env_file=ctx.obj.get('env_file'))
+    ctx.obj['use_cache'] = cache
+
+    build_container = _prepare_build_container(
```
Comment (ofir-amir) [inline skipper/cli.py:251]: pass ctx.obj to _prepare_build_container? you are using it in multiple places

File: tests/test_cli.py
Comment (ofir-amir) [inline tests/test_cli.py:1]: I this test config from skipper.yaml only or also from CLI or you expect to be handled by click?

**Review by ofir-amir — CHANGES_REQUESTED** [review]

File: skipper/cli.py (hunk 4 above)
Comment (khizunov) [reply skipper/cli.py:251]: IMO `_prepare_build_container` shouldn't be aware of ctx.

File: tests/test_cli.py
Comment (khizunov) [reply tests/test_cli.py:1]: @ofir-amir this test is testing builder functionality.

Comment (ofir-amir) [reply tests/test_cli.py:1]: oops - the question was on [tests/test_config.py](https://github.com/Stratoscale/skipper/pull/167/files#diff-da24b355349b53e4c8047034767d008e4c2ce9ea6c8933a3acd50d0828f00528)

Comment (khizunov) [reply tests/test_cli.py:1]: these tests are only for config file. CLI verified by test_cli

**Review by ofir-amir — APPROVED** [review]

**Pull request merged**